src/node/interfaces.cpp

```diff
@@ -458,17 +459,20 @@ class NotificationsProxy : public CValidationInterface
 
 class NotificationsHandlerImpl : public Handler
 {
+private:
+    CMainSignals& m_signals;
```

Comment:
Technically i think this reference can be dangling at `Shutdown()` (if there is still some wallets loaded) since we don't `reset()` the `node.wallet_loader` before resetting `node.main_signals`.

Reply:
Ah, right, the ownership of the loader is in `node.chain_clients`:
https://github.com/bitcoin/bitcoin/blob/65c05db660b2ca1d0076b0d8573a6760b3228068/src/wallet/init.cpp#L129-L131

So it's dropped before `node.main_signals` is.